snapcraft/plugins/v2/catkin.py

```diff
@@ -0,0 +1,105 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2020 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The catkin plugin for ROS 1 parts.
+
+    - catkin-packages:
+      (list of strings)
+      List of catkin packages to build. If not specified, all packages in the
+      workspace will be built. If set to an empty list ([]), no packages will
+      be built, which could be useful if you only want ROS debs in the snap.
+
+    - catkin-packages-ignore:
+      (list of strings)
+      List of catkin packages to ignore (i.e. not build or install). If not
+      specified or set to an empty list ([]), no packages will be ignored.
+
+    - catkin-cmake-args:
+      (list of strings)
+      Arguments to pass to cmake projects. Note that any arguments here which match
+      catkin arguments need to be prefixed with a space. This can be done by quoting
+      each argument with a leading space.
+
+    This plugin expects the build-environment `ROS_DISTRO` and `ROS_BUILD_BASE`
+    to be populated by the `ros-<distro>` extension.
```

Comment:
more that expects, I would say relies on those variables and they can be handled gracefully by the ros-<distro> extension or something along those lines.

Could be like browser support from the Netscape days: "works best with the ros-<distro> extension" :-)

Reply:
Indeed, rephrased.